stan/math/opencl/kernel_cl.hpp

```diff
@@ -25,20 +25,20 @@ namespace opencl_kernels {
  * Compile an OpenCL kernel.
  *
  * @param name The name for the kernel
- * @param source A string literal containing the code for the kernel.
+ * @param sources A std::vector of strings containing the code for the kernel.
  * @param options The values of macros to be passed at compile time.
- * @note The macros defined in kernels/helpers.hpp are included in the kernel
- *  compilation for ease of writing and reading kernels.
  */
-inline auto compile_kernel(const char* name, const char* source,
+inline auto compile_kernel(const char* name, std::vector<const char*> sources,
```

Comment:
Very elegant! Excellent :)

Reply:
Agree! I think Pretty sure we can make that
 `const std::vector<const char*>& sources`

Reply:
^ I think that's true most places. @t4c1 while you are here would you mind cleaning up the signatures for stuff we can make const &?

Reply:
Of course. I'm also making `std::map<const char*, int> options` a reference.